### ISSUE TITLE
Nextjs 9.3 Case sensitive issue with SSG route causes unexpected server error accessing a dynamic route the first time after being deployed to now

### ISSUE THREAD

**Issue by johnrjj** [vercel/next.js#10922]
# Bug report

## Describe the bug

Hey there, big fan of Next.js!

I'm using the dynamic route feature, along with SSG (getStaticPaths and getStaticProps) in 9.3. It has been working great (even through the 9.2 canary), but I ran into one problem:

I noticed a consistent bug between dynamic routes, getStaticPaths/getStaticProps, case sensitivity, and the router, where the site can crash when hitting a route with a different casing from the SSG'ed one provided after a fresh deploy on now. (App crash, with the following error message: An unexpected error has occurred.)

## To Reproduce

**UPDATE: I am able to not get it to repro when I move `[...slug.tsx]` into another directory (e.g. `pages/[...slug.tsx] --> pages/anotherfolder/[...slug.tsx]`. It appears to be an issue with having it at the root level directory.**

Here's the following repro:

You can find the code for the repro here: https://github.com/johnrjj/nextjs-9.3-case-sensitive-ssg-bug-repro  (it's just a create-next-app + a dynamic page at the root)-- If you deploy that to now, and try to go to `https://DEPLOYED_URL/foo` for the first time after deploying, the site will crash. However, if you redeploy and try again at `https://DEPLOYED_URL/FOO` it will work. The lowercase route will consistently crash after a deploy, but only once, after that it seems to fix itself.

- Have a page with a dynamic route (e.g. `[...slug.tsx]`)
- Give getStaticPaths a slug of ['FOO'] (i.e. /FOO as a route)
- Deploy to now.sh
- Try to access /foo (lowercase is when it repros, vs the seeded FOO uppercase)
- Get an unknown server error ONLY the first time you try to access the page. The following is found as an error in the console: 
```
SyntaxError: Unexpected token < in JSON at position 0
X @ main-32c9e20d835b8beaccb9.module.js:1
foo:1 Uncaught (in promise) SyntaxError: Unexpected token < in JSON at position 0
```

## Expected behavior

Not to crash the first time

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: 9.3

**Comment by johnrjj** [comment]
I am able to not get it to repro when I move `[...slug.tsx]` into another directory (e.g. `pages/[...slug.tsx] --> pages/anotherfolder/[...slug.tsx]`. It appears to be an issue with having it at the root level directory.

**Comment by johnrjj** [comment]
UPDATE: Now, regardless of folder, next.js does not respect being case-insensitive, but only when deployed to now.sh. Local build on OSX works fine.

**Comment by e-karma** [comment]
I'm having a similar issue, but isn't due to case-sensitive, but the presence of a home route with no path.  

This is in an attempt to dynamically load all routes from a CMS API by including getInitialProps on the _app file.  

Url.com/ => No loading & 'An unexpected error has occurred.'

When I changed the path to Url.com/home => It Loaded

**Comment by mufaddal7** [comment]
Facing the same issue, is there a timeline by when this bug will be solved? @Timer 

@johnrjj How are you working around till the bug is unsolved?

**Comment by balazsorban44** [comment]
This issue has been automatically closed because it wasn't verified against next@canary. If you think it was closed by accident, please leave a comment. If you are running into a similar issue, please open a new issue with a reproduction. Thank you.
